conans/test/integration/graph/test_skip_binaries.py

```diff
@@ -92,3 +92,22 @@ def test_build_scripts_no_skip():
     c.run("install app")
     c.assert_listed_binary({"script/0.1": ("da39a3ee5e6b4b0d3255bfef95601890afd80709", "Cache")},
                            build=True)
+
+
+def test_list_skip_printing():
+    """ make sure that when a package is required in the graph, it is not marked as SKIP, just
+    because some other part of the graph is skipping it. In this case, a tool_require might be
+    necessary for some packages building from soures, but not for others
+    """
+    c = TestClient()
+    c.save({"tool/conanfile.py": GenConanfile("tool", "0.1"),
+            "pkga/conanfile.py": GenConanfile("pkga", "0.1").with_tool_requires("tool/0.1"),
+            "pkgb/conanfile.py": GenConanfile("pkgb", "0.1").with_requires("pkga/0.1")
+                                                            .with_tool_requires("tool/0.1"),
+            "app/conanfile.py": GenConanfile().with_requires("pkgb/0.1")})
+    c.run("create tool")
+    c.run("create pkga")
+    c.run("export pkgb")
+    c.run("install app --build=missing")
+    c.assert_listed_binary({"tool/0.1": ("da39a3ee5e6b4b0d3255bfef95601890afd80709", "Cache")},
+                           build=True)
```

Comment:
@memsharded I just checked it out, and I think it's something wrong.

Running your test with your changes, this is the output:

```
======== Computing dependency graph ========
Graph root
    conanfile.py: /private/var/folders/s8/zgn806y90xq7bdh64hhfw9th0000gr/T/tmpk6fuo72kconans/path with spaces/app/conanfile.py
Requirements
    pkga/0.1#2961466d4a8f03bcc567de3f0250839e - Cache
    pkgb/0.1#c205d447673f68eeea2c3eece464f126 - Cache
Build requirements
    tool/0.1#2d65f1b4af1ce59028f96adbfe7ed5a2 - Cache

======== Computing necessary packages ========
Requirements
    pkga/0.1#2961466d4a8f03bcc567de3f0250839e:da39a3ee5e6b4b0d3255bfef95601890afd80709#0ba8627bd47edc3a501e8f0eb9a79e5e - Cache
    pkgb/0.1#c205d447673f68eeea2c3eece464f126:47a5f20ec8fb480e1c5794462089b01a3548fdc5 - Build
Build requirements
    tool/0.1#2d65f1b4af1ce59028f96adbfe7ed5a2:da39a3ee5e6b4b0d3255bfef95601890afd80709#0ba8627bd47edc3a501e8f0eb9a79e5e - Cache
```

Now, running it without your change, the output is mostly the same:

```
======== Computing dependency graph ========
Graph root
    conanfile.py: /private/var/folders/s8/zgn806y90xq7bdh64hhfw9th0000gr/T/tmpj_6amfbeconans/path with spaces/app/conanfile.py
Requirements
    pkga/0.1#2961466d4a8f03bcc567de3f0250839e - Cache
    pkgb/0.1#c205d447673f68eeea2c3eece464f126 - Cache
Build requirements
    tool/0.1#2d65f1b4af1ce59028f96adbfe7ed5a2 - Cache

======== Computing necessary packages ========
Requirements
    pkga/0.1#2961466d4a8f03bcc567de3f0250839e:da39a3ee5e6b4b0d3255bfef95601890afd80709#0ba8627bd47edc3a501e8f0eb9a79e5e - Cache
    pkgb/0.1#c205d447673f68eeea2c3eece464f126:47a5f20ec8fb480e1c5794462089b01a3548fdc5 - Build
Build requirements
    tool/0.1#2d65f1b4af1ce59028f96adbfe7ed5a2:da39a3ee5e6b4b0d3255bfef95601890afd80709#0ba8627bd47edc3a501e8f0eb9a79e5e - Cache
```

Should it appear as a `- Skip` message for `tool/0.1`? Am I missing anything?

Reply:
Maybe it is not fully deterministic, it depends on the graph evaluation order, which might be different in different OS?
If I remove the changes with:
```diff
--- a/conan/cli/printers/graph.py
+++ b/conan/cli/printers/graph.py
@@ -98,7 +98,7 @@ def print_graph_packages(graph):
                 existing = requires.setdefault(node.pref, [node.binary, node.binary_remote])
         # TO avoid showing as "skip" something that is used in other node of the graph
         if existing[0] == BINARY_SKIP:
-            existing[0] = node.binary
+            pass # existing[0] = node.binary

     def _format_requires(title, reqs_to_print):
         if not reqs_to_print:
```
Then the test clearly fails, because it prints the ``Skip`` instead of ``Cache``

Reply:
I have updated the test to make it fail if the fix is not in place, please @franramirez688 review